packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -170,7 +171,7 @@ export class TokenDetectionController extends BaseController<
       });
 
       if (isDetectionEnabledForNetwork && isChainIdChanged) {
-        this.detectTokens();
+        logRejection(this.detectTokens());
```

Comment:
We may as well `await` async calls in event handlers to get better stack traces if they fail